### PR TITLE
[AMORO-2496][DOC]: Add configuration docs for AMS terminal

### LINKED ISSUE #2496: [Improvement]:  Add configuration docs for AMS terminal
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

The current AMS terminal configurations lack documentation and require supplementation.

### How should we improve?

- Add configuration description in docs
- Add configuration examplex in configuration files

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by czy006**
## Why are the changes needed?

Close #2496.

## Brief change log

- Add configuration docs for AMS terminal

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (docs)

### CODE REVIEW

File: docs/user-guides/configurations.md [hunk 1]
```diff
@@ -134,3 +134,15 @@ If using Iceberg Format，please refer to [Iceberg configurations](https://icebe
...
+
+### Terminal configurations
+
+Terminal support local and kyuubi, the default is local. If the user uses local to run in the spark local context, you can set the **spark.*** configuration, and if you use kyuubi, you can set the **kyuubi.*** configuration
+
+| Key                      | Default | Description                                             |
+|--------------------------|---------|---------------------------------------------------------|
+| terminal.backend         | local   | terminal backend implement. local, kyuubi are supported |
+| terminal.factory         | -       | session factory implement of terminal                   |
+| terminal.result.limit    | 1000    | limit of result-set                                     |
```
Comment (majin1102) [inline docs/user-guides/configurations.md:146]: result-set could be resultset or result set.
Have not seen this before

File: docs/user-guides/configurations.md [hunk 2]
```diff
@@ -134,3 +134,15 @@ If using Iceberg Format，please refer to [Iceberg configurations](https://icebe
...
 | base.hive.auto-sync-data-write    | false            | Whether synchronize data changes of Hive Table from HMS, this should be true when writing to Hive      |
 | base.hive.consistent-write.enabled | true            | To avoid writing dirty data, the files written to the Hive directory will be hidden files and renamed to visible files upon commit. |
+
+### Terminal configurations
+
+Terminal support local and kyuubi, the default is local. If the user uses local to run in the spark local context, you can set the **spark.*** configuration, and if you use kyuubi, you can set the **kyuubi.*** configuration
+
+| Key                      | Default | Description                                             |
+|--------------------------|---------|---------------------------------------------------------|
+| terminal.backend         | local   | terminal backend implement. local, kyuubi are supported |
```
Comment (majin1102) [inline docs/user-guides/configurations.md:144]: implementation

File: docs/user-guides/configurations.md [hunk 3]
```diff
@@ -134,3 +134,15 @@ If using Iceberg Format，please refer to [Iceberg configurations](https://icebe
...
+| terminal.factory         | -       | session factory implement of terminal                   |
```
Comment (majin1102) [inline docs/user-guides/configurations.md:145]: I don't get this
Could you explain more?

File: docs/user-guides/configurations.md [hunk 4]
```diff
@@ -134,3 +134,15 @@ If using Iceberg Format，please refer to [Iceberg configurations](https://icebe
...
+| terminal.result.limit    | 1000    | limit of result-set                                     |
+| terminal.stop-on-error   | false   | stop script execution if any statement execute failed.  |
+| terminal.session.timeout | 30      | session timeout in minute                               |
```
Comment (majin1102) [inline docs/user-guides/configurations.md:148]: minutes

File: docs/user-guides/configurations.md [hunk 5]
```diff
@@ -134,3 +134,15 @@ If using Iceberg Format，please refer to [Iceberg configurations](https://icebe
...
+### Terminal configurations
+
+Terminal support local and kyuubi, the default is local. If the user uses local to run in the spark local context, you can set the **spark.*** configuration, and if you use kyuubi, you can set the **kyuubi.*** configuration
+
+| Key                      | Default | Description                                             |
+|--------------------------|---------|---------------------------------------------------------|
+| terminal.backend         | local   | terminal backend implement. local, kyuubi are supported |
+| terminal.factory         | -       | session factory implement of terminal                   |
+| terminal.result.limit    | 1000    | limit of result-set                                     |
+| terminal.stop-on-error   | false   | stop script execution if any statement execute failed.  |
```
Comment (majin1102) [inline docs/user-guides/configurations.md:147]: (Suggest) if stop all statements in one execution when any statement failed.

**Comment by majin1102** [comment]
Hi, I left some comments, PTAL

**Comment by zhoujinsong** [comment]
@czy006 Hi, thanks a lot for your contribution!

Now all configurations of AMS are described in `docs/admin-guides/deployment.md#Configuration`.
I think you should move these new descriptions into it.

And `docs/user-guides/configuration.md` is used to describe the configurations of tables now.

File: docs/user-guides/configurations.md (hunk 3 above)
Comment (czy006) [reply docs/user-guides/configurations.md:145]: **ArcticManagementConf.TERMINAL_SESSION_FACTORY** custom treminal implement

File: docs/user-guides/configurations.md (hunk 1 above)
Comment (czy006) [reply docs/user-guides/configurations.md:146]: **ArcticManagementConf.terminal.result.limit** it limit result callback rows

**Comment by czy006** [comment]
> @czy006 Hi, thanks a lot for your contribution!
> 
> Now all configurations of AMS are described in `docs/admin-guides/deployment.md#Configuration`. I think you should move these new descriptions into it.
> 
> And `docs/user-guides/configuration.md` is used to describe the configurations of tables now.

The new commit has been moved to the correct location

File: docs/admin-guides/deployment.md [hunk 6]
```diff
@@ -265,6 +265,18 @@ The following JVM options could be set in `${AMORO_CONF_DIR}/jvm.properties`.
 | jmx.remote.port | "-Dcom.sun.management.jmxremote.port=${value}  | Enable remote debug      |
 | extra.options   | "JAVA_OPTS="${JAVA_OPTS} ${JVM_EXTRA_CONFIG}"  | The addition jvm options |
 
+### Terminal configurations
+
+Terminal support local and kyuubi, the default is local. If the user uses local to run in the spark local context, you can set the **spark.*** configuration, and if you use kyuubi, you can set the **kyuubi.*** configuration
+
+| Key                      | Default | Description                                                        |
+|--------------------------|---------|--------------------------------------------------------------------|
+| terminal.backend         | local   | terminal backend implementation. local, kyuubi are supported       |
```
Comment (baiyangtx) [inline docs/admin-guides/deployment.md:274]:
```suggestion
| terminal.backend         | local   | Terminal backend implementation. local, kyuubi and custom are valid values.     |
```

File: docs/admin-guides/deployment.md [hunk 7]
```diff
@@ -265,6 +265,18 @@ The following JVM options could be set in `${AMORO_CONF_DIR}/jvm.properties`.
...
+| terminal.factory         | -       | session factory implement of terminal                              |
```
Comment (baiyangtx) [inline docs/admin-guides/deployment.md:275]:
```suggestion
| terminal.factory         | -       | Session factory implement of terminal, `terminal.backend` must be `custom` if this is set.                              |
```

File: docs/admin-guides/deployment.md [hunk 8]
```diff
@@ -265,6 +265,18 @@ The following JVM options could be set in `${AMORO_CONF_DIR}/jvm.properties`.
...
+| terminal.result.limit    | 1000    | limit of result-set                                                |
```
Comment (baiyangtx) [inline docs/admin-guides/deployment.md:276]:
```suggestion
| terminal.result.limit    | 1000    | Row limit of result-set                                                |
```

File: docs/admin-guides/deployment.md [hunk 9]
```diff
@@ -265,6 +265,18 @@ The following JVM options could be set in `${AMORO_CONF_DIR}/jvm.properties`.
...
+| terminal.stop-on-error   | false   | if stop all statements in one execution when any statement failed. |
+| terminal.session.timeout | 30      | session timeout in minutes.                                        |
```
Comment (baiyangtx) [inline docs/admin-guides/deployment.md:278]:
```suggestion
| terminal.session.timeout | 30      | Session timeout in minutes.                                        |
```

File: docs/admin-guides/deployment.md [hunk 10]
```diff
@@ -265,6 +265,18 @@ The following JVM options could be set in `${AMORO_CONF_DIR}/jvm.properties`.
...
+### Terminal configurations
+
+Terminal support local and kyuubi, the default is local. If the user uses local to run in the spark local context, you can set the **spark.*** configuration, and if you use kyuubi, you can set the **kyuubi.*** configuration
+
+| Key                      | Default | Description                                                        |
+|--------------------------|---------|--------------------------------------------------------------------|
+| terminal.backend         | local   | terminal backend implementation. local, kyuubi are supported       |
+| terminal.factory         | -       | session factory implement of terminal                              |
+| terminal.result.limit    | 1000    | limit of result-set                                                |
+| terminal.stop-on-error   | false   | if stop all statements in one execution when any statement failed. |
```
Comment (baiyangtx) [inline docs/admin-guides/deployment.md:277]:
```suggestion
| terminal.stop-on-error   | false   | When a statement fails to execute, stop execution or continue executing the remaining statements.  |
```

**Review by baiyangtx — COMMENTED** [review]
I left some comments, PTAL

**Comment by czy006** [comment]
> I left some comments, PTAL

Thanks for your contribution and correction. By the way, I extracted these parameter descriptions from **ArcticManagementConf**. Does this mean that these descriptions need to be modified as well?

**Comment by baiyangtx** [comment]
> > I left some comments, PTAL
> 
> Thanks for your contribution and correction. By the way, I extracted these parameter descriptions from **ArcticManagementConf**. Does this mean that these descriptions need to be modified as well?

Yes, I think these descriptions shoud be modified as well.

**Review by baiyangtx — APPROVED** [review]
LGTM

**Review by huyuanfeng2018 — APPROVED** [review]
+1

**Pull request merged**